### PR TITLE
[Perf] Batch KV scale host conversion

### PR DESCRIPTION
**Pull request by ShuhaoZhangTony**
## Summary

This PR reduces synchronization overhead when `--calculate-kv-scales` computes attention KV scales.

Instead of calling `.item()` separately for q/k/v scales, it batches the three scalar tensors into one host transfer via `torch.stack(...).to("cpu").tolist()`.

The same pattern is applied to:

- `Attention.calc_kv_scales`
- `MLAAttention.calc_kv_scales`

A focused unit test covers both regular attention and MLA scale updates, including the stored tensor scales and float mirror values.

## Why

Each `.item()` can force a separate device-to-host synchronization. These scale calculations happen in attention setup paths, so batching the host conversion avoids redundant sync points while preserving the existing public behavior.

## Validation

- `git diff --check`
- `python -m py_compile tests/model_executor/test_kv_scale_calc.py`

Attempted but could not run locally in this container:

- `python -m pytest tests/model_executor/test_kv_scale_calc.py`

The default Python environment is missing `torch`, and the available repo virtual environments are missing the needed test dependencies.

### CODE REVIEW

**Comment by ShuhaoZhangTony** [comment]
CI note: the current failing `pre-run-check` is the vLLM contributor gate. Please add the `ready` or `verified` label to trigger the full CI once the PR is ready for maintainer review.

**Comment by ShuhaoZhangTony** [comment]
Branch renamed to remove the codex prefix. Follow-up draft PR: #49017, rebased onto current main.
